### ISSUE TITLE
Uncaught TypeError: this.projectFile.read(...).then(...).then(...).then(...).then(...).catch(...)...

### ISSUE THREAD

**Issue by gtallen1187** [stamen/atom-cartodb#15]
Steps to reproduce:
1. Clone repo from https://github.com/stamen/cartodb-basemaps
2. Open in Atom
3. Ran "Cartodb: Preview" from the command palette.  Error:  could not find `project.yml` in the project root.
4. Changed name of `positron.yml` to `project.yml`
5. Received error below

**Atom Version**: 1.1.0
**System**: Microsoft Windows 7 Enterprise 
**Thrown From**: [cartodb](https://github.com/stamen/atom-cartodb) package, v0.7.2
### Stack Trace

Uncaught TypeError: this.projectFile.read(...).then(...).then(...).then(...).then(...).catch(...).finally is not a function

```
At /C:/Users/Allen%20Greg/.atom/packages/cartodb/lib/cartodb-preview-view.js:141

TypeError: this.projectFile.read(...).then(...).then(...).then(...).then(...).catch(...).finally is not a function
    at CartodbPreviewView.projectDidChange (C:/Users/Allen Greg/.atom/packages/cartodb/lib/cartodb-preview-view.js:111:15)
    at CartodbPreviewView.attached (C:/Users/Allen Greg/.atom/packages/cartodb/lib/cartodb-preview-view.js:149:10)
    at space-pen-div.View.element.attached (C:\Users\Allen Greg\AppData\Local\atom\app-1.1.0\resources\app.asar\node_modules\atom-space-pen-views\node_modules\space-pen\lib\space-pen.js:164:63)
    at space-pen-div.CustomElementPrototype.attachedCallback (C:\Users\Allen Greg\AppData\Local\atom\app-1.1.0\resources\app.asar\node_modules\atom-space-pen-views\node_modules\space-pen\lib\space-pen.js:51:55)
    at atom-pane.PaneElement.activeItemChanged (C:\Users\Allen Greg\AppData\Local\atom\app-1.1.0\resources\app.asar\src\pane-element.js:131:24)
    at Emitter.module.exports.Emitter.emit (C:\Users\Allen Greg\AppData\Local\atom\app-1.1.0\resources\app.asar\node_modules\event-kit\lib\emitter.js:86:11)
    at Pane.module.exports.Pane.setActiveItem (C:\Users\Allen Greg\AppData\Local\atom\app-1.1.0\resources\app.asar\src\pane.js:247:22)
    at Pane.module.exports.Pane.addItem (C:\Users\Allen Greg\AppData\Local\atom\app-1.1.0\resources\app.asar\src\pane.js:345:14)
    at Pane.module.exports.Pane.activateItem (C:\Users\Allen Greg\AppData\Local\atom\app-1.1.0\resources\app.asar\src\pane.js:310:14)
    at C:\Users\Allen Greg\AppData\Local\atom\app-1.1.0\resources\app.asar\src\workspace.js:474:16
```
### Commands

```
     -1:15.1.0 command-palette:toggle (atom-pane.pane.active)
     -1:11.1.0 cartodb:preview (atom-pane.pane.active)
     -0:22.3.0 tree-view:move (span.name.icon.icon-file-text)
     -0:18.3.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -0:16.2.0 command-palette:toggle (atom-pane.pane.active)
     -0:14.4.0 core:confirm (atom-text-editor.editor.mini.is-focused)
     -0:14.4.0 cartodb:preview (atom-pane.pane.active)
```
### Config

``` json
{
  "core": {
    "themes": [
      "atom-material-ui",
      "atom-material-syntax"
    ],
    "followSymlinks": true,
    "projectHome": "C:\\Users\\Allen Greg\\Documents\\GitHub",
    "disabledPackages": [
      "linter-bootlint",
      "linter-jshint",
      "atom-css-comb"
    ]
  },
  "cartodb": {
    "apiKey": "[[ redacted ]]",
    "username": "gallen"
  }
}
```
### Installed Packages

``` coffee
# User
atom-material-syntax, v0.2.4
atom-material-ui, v0.6.3
cartodb, v0.7.2
color-picker, v2.0.13
emmet, v2.3.13
git-tab-status, v1.9.2
grunt-runner, v0.11.0
language-carto, v0.6.2
linter, v1.9.1
pigments, v0.18.1
tool-bar, v0.1.9
tool-bar-almighty, v0.2.0

# Dev
No dev packages
```

**Comment by sethfitz** [comment]
Dupes #14 

This appears to be a regression in atom-1.1.0.  I'll have a fix out shortly.

**Comment by gtallen1187** [comment]
woops, sorry about the duplication!

**Comment by sethfitz** [comment]
No problem. Marking it was more for me than you. Have you had a chance to
try the updated version?

**Comment by gtallen1187** [comment]
nope not yet, but i will right now.

**Comment by gtallen1187** [comment]
So i'm admittedly a little new at working with basemaps like this, and when i just tried to run it again, i got an error telling me i'm missing a `layers.yml` file.

This is typically the point where i realize i should probably do some more reading and make sure i understand the project structure before i start bothering other people for help :grin: 

one last question: any good suggestions as to where to get started?

**Comment by sethfitz** [comment]
;-)

Here's a sample `layers.yml`: https://github.com/stamen/CartoDB-basemaps/blob/master/layers.yml

It's in the same repo that contains the style and layer definitions that this was implemented against (which will undoubtedly refer to tables you don't have).  Hopefully it's enough to get started.

There are a number of things where there are implicit defaults (typically matching the values of stuff) that should be documented, so keep posting questions here and I'll slowly get around to improving the docs (which largely assume that I've walked you through it in person at the moment ;-).
